edx_sga/static/js/src/edx_sga.js

```diff
@@ -314,10 +321,82 @@ function StaffGradedAssignmentXBlock(runtime, element) {
                     });
                 block.find('#staff-debug-info-button')
                     .leanModal();
+
+                $(element).find('#download-init-button').click(function(e) {
+                  e.preventDefault();
+                  var self = this;
+                  $(self).addClass("disabled");
+                  $.get(
+                    prepareDownloadSubmissionsUrl,
+                    function(data) {
+                      if (data["downloadable"]) {
+                        window.location = downloadSubmissionsUrl;
+                        $(element).find('.task-message').hide();
+                      } else {
+                        $(element).find('.task-message').show();
+                        $(element).find('.task-message').html(comeBackMessage());
+                        $(element).find('.task-message').removeClass("ready-msg");
+                        $(element).find('.task-message').addClass("preparing-msg");
+                      }
+                      $(self).removeClass("disabled");
```

Comment:
What I was saying in my previous comment is that the "Download" button should be disabled as long as we know that the submission file is processing. After a user clicks the "Download" button, and before the submission file finishes or experiences some error, we do not want the user clicking the "Download" button and making more requests to prepare the submission file. In this code, you are re-enabling the button as soon as the request to start processing the submission file comes back, even when the response indicates the the file is currently processing. When the submission file completes, or some error is experienced and the task is no longer running, that is when we can safely re-enable the "Download" button. Does that make sense?

Reply:
ok